UXPin/uxpin-merge-tools
# Fix vulnerabilities in Storybook deployer

## Goal
An attempt to fix "Moderate" issues reported by `yarn audit` command (`1` Moderate issues) and upgrade dependencies. 

The plan was to upgrade the vulnerable package and then update other packages. Packages that are shared in the `merge-cli` were upgraded to the same version used in the `merge-cli`. This was done to make it easier to move shared packages between storybook and merge CLIs to the root  in the near future.

- Upgrade vulnerable package `aws-sdk`
- Upgrade `types/mime-types`
- Upgrade `@types/node`
- Upgrade `@typescript-eslint/eslint-plugin`
- Upgrade `@typescript-eslint/parser`
- Upgrade `eslint`
- Upgrade `typescript`
- Upgrade `axios`
- Upgrade `chalk`
- Upgrade `commander`

This should also fix the warning issues when installing the package
![-zsh 2023-08-16 at 1 54 52 PM](https://github.com/UXPin/uxpin-merge-tools/assets/27636896/31eaddf8-4725-436d-bb30-d5ce321b0c33)


## How to test
A dev version of the package can be found [here](https://www.npmjs.com/package/@uxpin/storybook-deployer/v/1.0.1-dev-dev.149) `1.0.1-dev-dev.149`

- Can use this [library](https://github.com/uxpin-merge/storybook-material-ui) for testing
- Upgrade `@uxpin/storybook-deployer`
- Check for the warnings listed in the image above when installing
- Try the CLI push feature
- run `yarn audit` there should be no vulnerabilities from `@uxpin/storybook-deployer`


related issue: https://github.com/UXPin/uxpin-issues/issues/1680
